clients/spark/core/src/main/hadoop2/scala/io/treeverse/clients/conditional/S3ClientBuilder.scala

```diff
@@ -28,16 +30,49 @@ object S3ClientBuilder extends io.treeverse.clients.S3ClientBuilder {
         )
       } else None
 
+    initializeClient(configuration, credentialsProvider, region, bucket)
+  }
+
+  private def initializeClient(
+      config: ClientConfiguration,
+      credentialsProvider: Option[AWSCredentialsProvider],
+      region: String,
+      bucket: String
+  ): AmazonS3 = {
     val builder = AmazonS3ClientBuilder
       .standard()
-      .withClientConfiguration(configuration)
+      .withClientConfiguration(config)
       .withRegion(region)
-
     val builderWithCredentials = credentialsProvider match {
       case Some(cp) => builder.withCredentials(cp)
       case None     => builder
     }
+    val client = builderWithCredentials.build
+
+    if (validateClientCorrectness(client, bucket)) {
+      return client
+    }
+    val bucketRegion = getAWSS3Region(client, bucket)
```

Comment:
Can we add a log here and before line 53?  It can really help me debug connection failures if I know which path this code took!

Reply:
What kind of log?